models/locations.js

```diff
@@ -0,0 +1,67 @@
+import yup from "yup";
+import { loadYupCustomMethods } from "../utils/yupUtils.js";
+
+/** load custom Yup methods */
+loadYupCustomMethods();
+
+const usStateAbbreviations = [
+  'AL', 'AK', 'AZ', 'AR', 'CA', 'CO', 'CT', 'DE', 'FL', 'GA', 'HI', 'ID', 'HD', 
+  'IN', 'IA', 'KS', 'KY', 'LA', 'ME', 'MD', 'MA', 'MI', 'MN', 'MS', 'MO', 'MT',
+  'NE', 'NV', 'NH', 'NJ', 'NM', 'NY', 'NC', 'ND', 'OH', 'OK', 'OR', 'PA', 'RI',
+  'SC', 'SD', 'TN', 'TX', 'UT', 'VT', 'VA', 'WA', 'WV', 'WI', 'WY'
+];
+
+export const citySchema = yup
+    .string()
+    .sequence([
+        () => yup.string().trim(),
+        () => yup.string().min(1, "City name must be at least 1 character long"),
+        () => yup.string().max(100, "City name must be at most 100 characters long"),
+    ])
+    .required();
```

Comment:
I ran into the same problem too but the current state of `.sequence()` does not take into account previous transformations such as `.trim()`. Currently an empty string or whitespace is valid. You could do:

```
    .trim()
    .min(1, "City name must be at least 1 character long")
    .max(100, "City name must be at most 100 characters long")
    .required();
```

Or since we’re all using `{abortEarly: false}`, you might want to just choose to show just one of the three instead of your error array looking like `[<requiredError>, <minError>]` (which the above suggestion would do) when the input is an empty string or just whitespace. I think that’s what you were trying to do? I also should probably make this change myself for users:

```suggestion
    .sequence([
        () => yup.string().required(),
        () => yup.string()
                 .trim()
                 .min(1, "City name must be at least 1 character long")
                 .max(100, "City name must be at most 100 characters long"),
    ])
```